code/modules/mob/living/simple_animal/hostile/terror_spiders/purple.dm

```diff
@@ -0,0 +1,88 @@
+
+// --------------------------------------------------------------------------------
+// ----------------- TERROR SPIDERS: T2 PURPLE TERROR -----------------------------
+// --------------------------------------------------------------------------------
+// -------------: ROLE: guarding queen nests
+// -------------: SPECIAL: chance to stun on hit
+// -------------: TO FIGHT IT: shoot it from range, bring friends!
+// -------------: SPRITES FROM: FoS, http://nanotrasen.se/phpBB3/memberlist.php?mode=viewprofile&u=386
+
+/mob/living/simple_animal/hostile/poison/terror_spider/purple
+	name = "Purple Terror spider"
+	desc = "An ominous-looking purple spider. It looks about warily, as if waiting for something."
+	spider_role_summary = "Guards the nest of the Queen of Terror."
+
+	icon_state = "terror_purple"
+	icon_living = "terror_purple"
+	icon_dead = "terror_purple_dead"
+	maxHealth = 200
+	health = 200
+	melee_damage_lower = 15
+	melee_damage_upper = 25
+	move_to_delay = 6
+
+	spider_tier = TS_TIER_2
+	spider_opens_doors = 2
+
+	ventcrawler = 0
+	environment_smash = 3
+
+	var/dcheck_counter = 0
+	var/queen_visible = 1
+	var/cycles_noqueen = 0
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/purple/spider_specialattack(mob/living/carbon/human/L, poisonable)
+	if(cycles_noqueen < 6 && prob(10))
+		visible_message("<span class='danger'>[src] rams into [L], knocking them to the floor!</span>")
+		L.Weaken(5)
+		L.Stun(5)
+	else
+		..()
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/purple/Life()
+	. = ..()
+	if(.) // if mob is NOT dead
+		if(!degenerate && spider_myqueen)
+			if(dcheck_counter >= 10)
+				dcheck_counter = 0
+				purple_distance_check()
+			else
+				dcheck_counter++
+	return .
```

Comment:
This is redundant- `.` is the default return value of the proc, it is automatically returned